File: backend/app/schemas/items.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class ItemCreate(BaseModel):
    tipo: str
    brand: str
    model: str
    revenda: str
    date_registered: str  # dd/mm/yyyy
    # Campos opcionais (variam por tipo)
    identificador: Optional[str] = None
    nota_fiscal: Optional[str] = None
    fornecedor: Optional[str] = None
    dominio: Optional[str] = None
    host: Optional[str] = None
    endereco_fisico: Optional[str] = None
    cpu: Optional[str] = None
    ram: Optional[str] = None
    storage: Optional[str] = None
    sistema: Optional[str] = None
    licenca: Optional[str] = None
    anydesk: Optional[str] = None
    setor: Optional[str] = None
    ip: Optional[str] = None
    mac: Optional[str] = None
    fornecedor: Optional[str] = None
    potencia_nominal: Optional[str] = None
    autonomia_estimada: Optional[str] = None
    ip_snmp: Optional[str] = None
    codigo_patrimonial: Optional[str] = None
    responsavel: Optional[str] = None
    local_instalacao: Optional[str] = None
    poe: Optional[str] = None
    quantidade_portas: Optional[str] = None


class ItemUpdate(BaseModel):
    tipo: Optional[str] = None
    brand: Optional[str] = None
    model: Optional[str] = None
    revenda: Optional[str] = None
    identificador: Optional[str] = None
    nota_fiscal: Optional[str] = None
    fornecedor: Optional[str] = None
    dominio: Optional[str] = None
    host: Optional[str] = None
    endereco_fisico: Optional[str] = None
    cpu: Optional[str] = None
    ram: Optional[str] = None
    storage: Optional[str] = None
    sistema: Optional[str] = None
    licenca: Optional[str] = None
    anydesk: Optional[str] = None
    setor: Optional[str] = None
    ip: Optional[str] = None
    mac: Optional[str] = None
    potencia_nominal: Optional[str] = None
    autonomia_estimada: Optional[str] = None
    ip_snmp: Optional[str] = None
    codigo_patrimonial: Optional[str] = None
    responsavel: Optional[str] = None
    local_instalacao: Optional[str] = None
    poe: Optional[str] = None
    quantidade_portas: Optional[str] = None


class ItemResponse(BaseModel):
    id: int
    tipo: Optional[str]
    brand: Optional[str]
    model: Optional[str]
    identificador: Optional[str]
    nota_fiscal: Optional[str]
    status: Optional[str]
    assigned_to: Optional[str]
    cpf: Optional[str]
    revenda: Optional[str]
    dominio: Optional[str]
    host: Optional[str]
    endereco_fisico: Optional[str]
    cpu: Optional[str]
    ram: Optional[str]
    storage: Optional[str]
    sistema: Optional[str]
    licenca: Optional[str]
    anydesk: Optional[str]
    setor: Optional[str]
    ip: Optional[str]
    mac: Optional[str]
    fornecedor: Optional[str]
    potencia_nominal: Optional[str]
    autonomia_estimada: Optional[str]
    ip_snmp: Optional[str]
    codigo_patrimonial: Optional[str]
    responsavel: Optional[str]
    local_instalacao: Optional[str]
    poe: Optional[str]
    quantidade_portas: Optional[str]
    date_registered: Optional[datetime]
    date_issued: Optional[datetime]
    peripheral_count: Optional[int] = 0

    class Config:
        from_attributes = True

File: frontend/src/api/items.ts
import api, { type Paginated } from './client'

export interface Item {
  id: number
  tipo?: string
  brand?: string
  model?: string
  identificador?: string
  nota_fiscal?: string
  status?: string
  assigned_to?: string
  cpf?: string
  revenda?: string
  dominio?: string
  host?: string
  endereco_fisico?: string
  cpu?: string
  ram?: string
  storage?: string
  sistema?: string
  licenca?: string
  anydesk?: string
  setor?: string
  ip?: string
  mac?: string
  fornecedor?: string
  potencia_nominal?: string
  autonomia_estimada?: string
  ip_snmp?: string
  codigo_patrimonial?: string
  responsavel?: string
  local_instalacao?: string
  poe?: string
  quantidade_portas?: string
  date_registered?: string
  date_issued?: string
  peripheral_count?: number
}

export interface ListItemsParams {
  tipo?: string
  status?: string
  revenda?: string
  search?: string
  limit?: number
  offset?: number
}

/**
 * `GET /api/items` paginado: aceita `tipo`/`status`/`revenda`/`search` e
 * `limit`/`offset`, devolvendo `{ items, total }`. Os filtros são aplicados no
 * SQL, não no cliente.
 *
 * Atenção ao omitir `limit`: o backend assume DEFAULT_PAGE_SIZE (50). Telas que
 * usam a lista para preencher um select precisam passar um limite explícito,
 * ou truncam em silêncio.
 */
export async function listItemsPaginated(params?: ListItemsParams): Promise<Paginated<Item>> {
  const res = await api.get('/items', { params })
  return res.data as Paginated<Item>
}

export async function getItem(id: number) {
  const res = await api.get(`/items/${id}`)
  return res.data as Item
}

export async function createItem(data: Record<string, unknown>) {
  const res = await api.post('/items', data)
  return res.data as { detail: string; id: number }
}

export async function updateItem(id: number, data: Record<string, unknown>) {
  const res = await api.put(`/items/${id}`, data)
  return res.data
}

export async function removeItem(id: number, reason: string, attachment?: File) {
  const form = new FormData()
  form.append('reason', reason)
  if (attachment) form.append('attachment', attachment)
  const res = await api.delete(`/items/${id}`, { data: form })
  return res.data
}
